### DISCUSSION TITLE
Reloading Job Schedules on-the-fly

### DISCUSSION THREAD

**Discussion by MattBax** [quartznet/quartznet#2325] [Q&A] [answered]
Hi folks,

We have a .NET 6 worker application that uses Quartz.NET. We store and retrieve our job names and schedules from Azure App Configuration (or appsettings.json when run on localhost). The Quartz Scheduler implementation is static and the job configuration is initalised once at start-up.

While it's possible for us to reload/refresh the Azure App Configuration "on-the-fly" using a Sentinel Key, we're unable to reload/refresh the updated job schedules into the Quartz Scheduler without restarting the application.

Does Quartz.NET provide a way to do this, without having to restart the application? I'm guessing that it's not possible while our Quartz Scheduler implementation is static, so we might need to refactor? I'd appreciate any helpers or documentation regarding this query.

Thanks in advance!

**Comment by lahma** [comment, marked as answer]
Currently it's not possible. Quartz supports on-the fly updates via XML configuration (it's configured to be  scanned for changes) and of course database backed storage where Quartz will query for latest changes.

Might not be impossible to achive with some changes though as `ContainerConfigurationProcessor` which handles DI configuration is based on `XMLSchedulingDataProcessor` which again is basis for the XML based updates (via `XMLSchedulingDataProcessorPlugin`).

**Reply by MattBax** [reply]
Thanks @lahma ! We'll look into using XML configuration.
